rules/macos/exfiltration_compress_credentials_keychains.toml

```diff
@@ -0,0 +1,44 @@
+[metadata]
+creation_date = "2020/08/14"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/08/14"
+
+[rule]
+author = ["Elastic"]
+description = """
+Adversaries may collect the keychain storage data from a system to acquire credentials. Keychains are the built-in way
+for macOS to keep track of users' passwords and credentials for many services and features such as WiFi passwords,
+websites, secure notes, certificates, and Kerberos.
+"""
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Preparation of Keychain crednetials exfiltration"
+references = ["https://objective-see.com/blog/blog_0x25.html"]
+risk_score = 73
+rule_id = "96e90768-c3b7-4df6-b5d9-6237f8bc36a8"
+severity = "high"
+tags = ["Elastic", "MacOS"]
+type = "query"
+
+query = '''
+event.category:process and not event.type:end and
+  process.name:(zip or tar or gzip or 7za or hdiutil) and
+  process.args:("/Library/Keychains/" or "/Network/Library/Keychains/" or "~/Library/Keychains/")
```

Comment:
This would miss compression of individual files within these directories, unless a wildcard is added. Even with the wildcards, if the working directory is within any of these or their sub-directories, it would also miss compression of the file names. 

Auditbeat has `process.working_directory`, but that is not currently populated by the endpoint agent

Reply:
Thats fair, incremental detection is a good thing. Perhaps changing the rule name to `Compression of Keychain Credentials Directories` would make sense too then

Reply:
changed